pkg/rhcos/builds.go

```diff
@@ -36,42 +35,77 @@ type metadata struct {
 	OSTreeVersion string `json:"ostree-version"`
 }
 
-func fetchLatestMetadata(ctx context.Context, channel string) (metadata, error) {
-	build := buildName
-	var err error
-	if build == "" {
-		build, err = fetchLatestBuild(ctx, channel)
-		if err != nil {
-			return metadata{}, errors.Wrap(err, "failed to fetch latest build")
+// rhcosBuildID is a version number e.g. 410.8.20180312.0
+var rhcosBuildID string
+
+// rhcosBuildChannel is a name for a stream of builds
+var rhcosBuildChannel string
+
+func fetchRHCOSBuild(ctx context.Context) (metadata, error) {
+	// Example uses:
```

Comment:
we have an overrides for OS:
https://github.com/openshift/installer/blob/3756253bf1b1ca715d878d8eb46c3f983ddfb1cc/hack/build.sh#L49-L52
https://github.com/openshift/installer/blob/3756253bf1b1ca715d878d8eb46c3f983ddfb1cc/pkg/asset/rhcos/image.go#L42-L46

why do we need more?

Reply:
Dup of this thread https://github.com/openshift/installer/pull/1423#discussion_r267412857